codegen/sr.rs

```diff
@@ -280,51 +271,192 @@ pub fn gen_sr_type_creation(grammar: &structs::Grammar) -> String {
     tokens.to_string()
 }
 
+fn lift_operand_simple(iter: &Ident, operand: &structs::Operand) -> TokenStream {
+    let kind_ident = Ident::new(&operand.kind, Span::call_site());
+    match operand.kind.as_str() {
+        "PairLiteralIntegerIdRef" |
+        "PairIdRefLiteralInteger" |
+        "PairIdRefIdRef" => quote! {
+        },
+        _ => {
+            let value = match operand.name.trim_matches('\'') {
+                "Length" => quote! { Token::new(*value) },
+                "Entry Point" => quote! { Token::new(*value) },
+                "Interface" => quote! { Token::new(*value) },
+                _ => quote! { value.clone() },
+            };
+            quote! {
+                match #iter.next() {
+                    Some(&mr::Operand::#kind_ident(ref value)) => Some(#value),
+                    Some(_) => Err(OperandError::Wrong)?,
+                    None => None,
+                }
+            }
+        }
+    }
+}
+
+fn lift_operand_complex(iter: &Ident, operand: &structs::Operand) -> TokenStream {
+    let value_token = lift_operand_simple(iter, operand);
+    match operand.quantifier.as_str() {
+        "" => quote! {
+            (#value_token).ok_or(OperandError::Missing)?
+        },
+        "?" => value_token,
+        "*" => quote! {
+            {
+                let mut vec = Vec::new();
+                while let Some(value) = #value_token {
+                    vec.push(value);
+                }
+                vec
+            }
+        },
+        other => panic!("wrong quantifier: {}", other),
+    }
+}
+
+pub fn gen_sr_structs_and_lifts(grammar: &structs::Grammar) -> (String, String) {
```

Comment:
Can we have some comments regarding this function?

(I wasn't doing good on this previously given it's just me playing with this repo. Now it's intended to be developed by more people so some documentation could really save newcomers efforts in understanding existing codebase. :)